tools/custom-action/Dockerfile

```diff
@@ -0,0 +1,59 @@
+FROM ubuntu:23.04
+
+ENV LANG=en_US.UTF-8
+
+RUN \
+  export DEBIAN_FRONTEND=noninteractive && \
+  apt update -y && \
+  apt install -y locales && \
+  locale-gen en_US.UTF-8 && \
+  sysctl -w kernel.unprivileged_userns_clone=1 && \
+  apt install -y \
+    nodejs \
+    npm \
+    git \
+    sudo \
+    libgtk-3-dev \
+    libnotify-dev \
+    libgconf-2-4 \
+    libnss3 \
+    libxss1 \
+    libasound2 \
+    python3-full \
+    python3-pip \
+    rsync
+
+RUN \
+  export DEBIAN_FRONTEND=noninteractive && \
+  python3 -m pip install --break-system-packages \
+    bikeshed==3.13.1 \
+    tree_sitter==0.20.1 && \
```

Comment:
One of the goals of the install-dependencies script is to have just one central place in the repository that sets which versions of all of our dependencies we use\*. Can we somehow avoid having an extra place to upgrade?

And if we can't, we need to comment both places telling maintainers to update the other one at the same time.

\* actually I noticed I already violated this in invoke-mermaid.sh but I don't know how to fix that.

Reply:
@kainino0x I have also thought about this a bit. The Python script in WGSL directory also violates by annotating version for package on its own in json file. I think that after this PR, maybe we can create an environment file where these are made into variables and then all scripts, Bash or Python or JavaScript, refer to it, also helping with redundancy of global installs.

Reply:
> The Python script in WGSL directory also violates by annotating version for package on its own in json file.

That's sort of true but those dependencies don't have pinned versions anywhere else in the repository. So at least there's only one source of truth. Except tree_sitter and tree-sitter-cli might need to stay somewhat in sync? But they're still two separate things.

Environment file sounds excellent.

Reply:
@kainino0x latest commit includes the environment file approach with proper paths!